src/Microsoft.Azure.WebJobs.Logging/Internal/LogReader.cs

```diff
@@ -132,6 +156,93 @@ public LogReader(CloudTable table)
             return new Segment<ActivationEvent>(l.ToArray(), null);
         }
 
+        // Epochs may be sparse. 
+        // We may frequently start an enumeration at DateTime.MaxValue
+        class EpochTableIterator
```

Comment:
This class needs to be cleaned up. And the SafeExecuteQuerySegmentedAsync specifically needs to be optimized to use continuation tokens. But that's just a contained implementation detail and I wanted to share the PR out to give a chance for feedback on the rest of this change.